geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,500 @@
+        goodcomps : list of str
+            * List of full paths to all "good" successful comparisons
+              (output and compare images match)
+            * Each str is prepended with a "compare_string" tag to identify which
+              comparison type was performed.
+        badcomps : list of str
+            * List of full paths to all "bad" unsuccessful comparisons
+              (output and compare images differ)
+            * Each str is prepended with a "compare_string" tag to identify which
+              comparison type was performed.
+        compare_strings : list of str
+            * List of all comparison "tags" included in goodcomps and badcomps lists.
+            * This list is used to remove the comparison tags from goodcomps and
+              badcomps to retrieve only the file path.
+
+        Returns
+        -------
+        goodcomps: list of str
+            All current good comparisons appended to the list passed in.
+        badcomps: list of str
+            All current bad comparisons appended to the list passed in.
+        compare_strings: list of str
+            All current comparison "tags" added to the list passed in.
+
+        Raises
+        ------
+        TypeError
+            Raised when current output product does not have an associated
+            comparison test defined.
+        """
+        matched_one = False
+        if self.name == "image_checker":
+            matched_one = True
+            compare_strings += ["IMAGE "]
+            if self.module.outputs_match(self, output_product, compare_product):
+                goodcomps += ["IMAGE {0}".format(output_product)]
+            else:
+                badcomps += ["IMAGE {0}".format(output_product)]
+
+        if self.name == "geotiff_checker":
+            matched_one = True
+            compare_strings += ["GEOTIFF "]
+            if self.module.outputs_match(output_product, compare_product):
+                goodcomps += ["GEOTIFF {0}".format(output_product)]
+            else:
+                badcomps += ["GEOTIFF {0}".format(output_product)]
+
+        if self.name == "text_checker":
+            matched_one = True
+            compare_strings += ["TEXT "]
+            if self.module.outputs_match(self, output_product, compare_product):
+                goodcomps += ["TEXT {0}".format(output_product)]
+            else:
+                badcomps += ["TEXT {0}".format(output_product)]
+
+        if self.name == "netcdf_checker":
+            matched_one = True
+            compare_strings += ["GEOIPS NETCDF "]
+            if self.module.outputs_match(self, output_product, compare_product):
+                goodcomps += ["GEOIPS NETCDF {0}".format(output_product)]
+            else:
+                badcomps += ["GEOIPS NETCDF {0}".format(output_product)]
+
+        if not matched_one:
+            raise TypeError(f"MISSING TEST for output product: {output_product}")
+
+        return goodcomps, badcomps, compare_strings
```

Comment:
I think we should just require "outputs_match" method on all output checker plugins, then we do not have to have the output checker plugin names listed here explictily.

The only unique part of each of these calls to outputs_match is the prefix to the goodcomps/badcomps strings - perhaps we could just use self.name.upper().  If that happens to break anything downstream with parsing the logs or something, we can fix that (I don't think it will have any impact).